### ISSUE TITLE
Schema Metadata Cache is always activated

### ISSUE THREAD

**Issue by ravage84** [cakephp/cakephp#14004]
* CakePHP Version: 3.8.7

It seems ``SchemaCache::_getSchema()`` always uses an activated cache, even when disabled through setting ``$config['cacheMetadata']`` to  ``false`` or ``0``.

https://github.com/cakephp/cakephp/blob/ed758075d1a12abebcc958baa450c49961c81dea/src/Database/SchemaCache.php#L103-L106

Proof:
https://3v4l.org/3p4ph

This line was likey copied from ``OrmCacheShell::_getSchema()`` in PR https://github.com/cakephp/cakephp/pull/10977

https://github.com/cakephp/cakephp/pull/10977/files#diff-2e213f8f3145bfa87b3f292c3bb2f2dcL109-L113.

But I'm wondering what these lines actually want to achieve.

Probably, is should be:

````php
$config = $connection->config();
if (!isset($config['cacheMetadata'])) {
    $connection->cacheMetadata(true);
}
````

This way, the cache would be activated when not set deliberately.

May be @burzum  can shed some light?

Found by @odan

**Comment by ADmad** [comment]
`SchemaCache` is only used by `SchemaCacheShell` which is meant to build/clear schema cache. It probably force enables metadata caching for the connection as other it won't make much to use the shell would it?

**Comment by othercorey** [comment]
@ravage84 Can we fix this for 3.9?
